ibis/expr/datatypes.py

```diff
@@ -93,22 +94,17 @@ def equals(self, other, cache=None):
     def _equal_part(self, other, cache=None):
         return True
 
-    def issubtype(self, parent):
-        return issubtype(self, parent)
-
     def castable(self, target, **kwargs):
         return castable(self, target, **kwargs)
 
     def cast(self, target, **kwargs):
         return cast(self, target, **kwargs)
 
     def scalar_type(self):
-        import ibis.expr.types as ir
-        return getattr(ir, '{}Scalar'.format(self.name))
+        return partial(self.scalar, dtype=self)
```

Comment:
@cpcloud These are actually "factories" not types. Are You OK with these?

Reply:
Yes, that's fine with me.